lib/SILOptimizer/PassManager/PassPipeline.cpp

```diff
@@ -347,6 +347,12 @@ void addFunctionPasses(SILPassPipelinePlan &P,
     P.addCOWArrayOpts();
     P.addDCE();
     P.addSwiftArrayPropertyOpt();
+    
+    // This string optimization can catch additional opportunities, which are
+    // exposed once optimized String interpolations (from the high-level string
+    // optimization) are cleaned up. But before the mid-level inliner inlines
+    // semantic calls.
+    P.addStringOptimization();
```

Comment:
What type of clean up is required ? Is this something that can be implemented in `InstSimplifier` in the future, so that we don't need another instance of the pass here ?

Reply:
All kind of function passes (e.g. SILMem2Reg) are needed to cleanup the code produced by the first StringOptimization.
Of course, we could put all optimizations in SILCombine, but that would defeat the purpose of having separate passes for separate optimizations. It's a design decision.

Reply:
Okay